tests/roles/pcp_cleanup/tasks/main.yaml

```diff
@@ -2,28 +2,40 @@
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
+    # Cleanup OpenStackControlPlane object
+    oc delete --ignore-not-found=true osctlplane -l core.openstack.org/openstackcontrolplane -n openstack
 
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack-nova-compute-ffu
-    oc delete --ignore-not-found=true --wait=false openstackcontrolplane/openstack
-    oc patch openstackcontrolplane openstack --type=merge --patch '
-    metadata:
-      finalizers: []
-    ' || true
-
-    while oc get pod | grep rabbitmq-server-0; do
-        sleep 2
-    done
-    while oc get pod | grep openstack-galera-0; do
-        sleep 2
+    # Ensure that all pods in openstack namespace are deleted
+    while oc get pod -n openstack | grep -E 'rabbitmq-server-0|openstack-galera-0'; do
+      sleep 2;
     done
 
+    # Delete Control Plane CRD
+    # oc delete crd --ignore-not-found=true openstackcontrolplanes.core.openstack.org
+
+    # Cleanup OpenStackDataplane objects
+    oc delete --ignore-not-found=true OpenStackDataPlaneDeployment --all -n openstack
+    oc delete --ignore-not-found=true OpenStackDataPlaneNodeSet --all -n openstack
+    oc delete --ignore-not-found=true OpenStackDataPlaneService --all -n openstack
+
+    # Delete Adoption pods
     oc delete --ignore-not-found=true --wait=false pod mariadb-copy-data
     oc delete --ignore-not-found=true --wait=false pvc mariadb-data
     oc delete --ignore-not-found=true --wait=false pod ovn-copy-data
-    oc delete --ignore-not-found=true secret osp-secret
-    oc delete issuer rootca-internal --ignore-not-found
-    oc delete secret rootca-internal --ignore-not-found
+
+    # Delete secrets
```

Comment:
This seems like it would need constant updating whenever there are new secrets. Can we not list the secrets and then loop through them?

Reply:
The way out of this is to delete the namespace so then all the resources in the namespace is deleted. Today the feeling is that deleting the namespace is not always works properly. Personally I think that is a bug and should be fixed. (And we fixed issues before in this area). But I don't have the details to how namespace deletion fails (It worked for me last time when I tried locally) so I cannot file a Jira bug tracking it.

Reply:
@holser sure. Thanks for the bug report OSPRH-10262 Please keep sending bugs when you see a stuck namespace deletion.